src/Documenter.jl

```diff
@@ -736,4 +751,74 @@ function generate(pkgname::AbstractString; dir=nothing)
     nothing
 end
 
+"""
+    external(path; target) -> ExternalPage
+
+Mark `path` as an external page. This allows us to use files outside of the documentation
+directory in [`makedocs`](@ref). Note that any relative links in these pages will end up
+broken.
+
+# Arguments
+- `path::AbstractString`: Path of the external file, relative to the `repo_root` keyword
+  argument to [`makedocs`](@ref).
+- `target::Union{AbstractString, Void}=nothing`: Destination of the external file, relative
+  to the docs source directory.
+
+# Usage
+```julia
+makedocs(;
+    ...,
+    pages=[
+        "Home" => external("README.md"; target="index.md"),
+    ],
+)
+```
+"""
+function external(path::AbstractString; target::Union{AbstractString, Void}=nothing)
+    if target == nothing && !endswith(path, ".md")
+        error("External file $path must have a Markdown extension or target must be set")
+    end
+    if target != nothing && !endswith(target, ".md")
+        error("External file target $target must have a Markdown extension")
+    end
+    return Documents.ExternalPage(path, target)
+end
+
+"""
+    copy_external!(pages, repo_root, dest_dir) -> Vector
+
+Copy all external files in `pages` into the documentation source directory and modify the
+input vector, replacing any [`Documents.ExternalPage`](@ref)s with their new paths.
+Returns the paths of any external pages to be deleted at the end of the build process.
+"""
+function copy_external!(pages::Vector, repo_root::AbstractString, dest_dir::AbstractString)
+    function copypage(page::Documents.ExternalPage)
+        source = joinpath(repo_root, page.path)
+        isfile(source) || error("External page $source was not found")
+        dest = joinpath(dest_dir, get(page.target, basename(page.path)))
+        isfile(dest) && error("A page called $(basename(dest)) already exists")
+        cp(source, dest)
+        return relpath(dest, dest_dir)
+    end
+
+    externals = String[]
+    for (i, page) in enumerate(pages)
+        if isa(page, Documents.ExternalPage)
+            pages[i] = copypage(page)
+            push!(externals, pages[i])
+        elseif isa(page, Pair) && isa(page.second, Documents.ExternalPage)
+            pages[i] = pages[i].first => copypage(page.second)
+            push!(externals, pages[i].second)
+        elseif isa(page, Pair) && isa(page.second, Vector)
+            for (j, el) in enumerate(page.second)
+                if isa(el, Documents.ExternalPage)
+                    pages[i].second[j] = copypage(el)
+                    push!(externals, pages[i].second[j])
+                end
+            end
```

Comment:
This only goes down two levels, right? `pages` can (in principle) have an arbitrary number of levels, so I think this should be done recursively. Something like how `walk_navpages` does it in `Builder.jl`.

Reply:
Yeah, I just wasn't aware that going deeper was allowed.